### PR TITLE
Add ik benchmark to ASV

### PR DESCRIPTION
**Pull request by Kenny-Vilella**
## Description
ik_benchmark example has been removed, this moved it to a proper asv benchmark.

## Checklist

- [ ] New or existing tests cover these changes
- [ ] The documentation is up to date with these changes
- [ ] `CHANGELOG.md` has been updated (if user-facing change)

## Test plan
Run standalone:
```
python asv/benchmarks/simulation/bench_ik.py --bench FastSolve
```
Run asv benchmark suite:
```
asv run --launch-method spawn HEAD^! --bench Fast
```

## Summary by CodeRabbit

* **New Features**
  * Added a complete inverse-kinematics benchmarking suite for the Franka Emika Panda, including a high-throughput CUDA-accelerated benchmark (FastIKSolve) and a runnable benchmark harness.

* **Chores**
  * Added utilities to generate IK problems and targets, configure and warm up the solver with CUDA graph replay, and evaluate solution success after runs.

### CODE REVIEW

File: asv/benchmarks/simulation/bench_ik.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            )
+            self.rot_obj[ee].set_target_rotations(
+                wp.array(self.tgt_r[:, ee].astype(np.float32, copy=False), dtype=wp.vec4)
+            )
+
+        # Capture CUDA graph
+        with wp.ScopedCapture() as cap:
+            self.solver.step(self.seeds_d, self.winners_d, iterations=self.ITERATIONS, step_size=self.STEP_SIZE)
+        self.solve_graph = cap.graph
+
```
[the remark this answers is not in the document]
Comment (Kenny-Vilella) [reply asv/benchmarks/simulation/bench_ik.py]: Done

File: asv/benchmarks/benchmark_ik.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                w1 * x2 + x1 * w2 + y1 * z2 - z1 * y2,
+                w1 * y2 - x1 * z2 + y1 * w2 + z1 * x2,
+                w1 * z2 + x1 * y2 - y1 * x2 + z1 * w2,
+            ),
+            axis=-1,
+        )
+
+    tgt_conj = np.concatenate([tgt_rot[..., :1], -tgt_rot[..., 1:]], axis=-1)
+    rel = _qmul(tgt_conj, bq[:, ee, 3:7])
+    rot_err = (2 * np.arctan2(np.linalg.norm(rel[..., 1:], axis=-1), np.abs(rel[..., 0]))).max(axis=-1)
```
[the remark this answers is not in the document]
Comment (Kenny-Vilella) [reply asv/benchmarks/benchmark_ik.py]: Done

File: asv/benchmarks/benchmark_ik.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    pos, rot = [], []
+    for q in q_batch:
+        wp.copy(model.joint_q, wp.array(q, dtype=wp.float32))
+        newton.eval_fk(model, model.joint_q, model.joint_qd, state)
+        body_q = state.body_q.numpy()
+        pos.append(body_q[list(ee_links), :3])
+        rot.append(body_q[list(ee_links), 3:7])
+    return np.stack(pos), np.stack(rot)
+
+
```
[the remark this answers is not in the document]
Comment (Kenny-Vilella) [reply asv/benchmarks/benchmark_ik.py]: Done

File: asv/benchmarks/simulation/bench_ik.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            self.pos_obj[ee].set_target_positions(
+                wp.array(self.tgt_p[:, ee].astype(np.float32, copy=False), dtype=wp.vec3)
+            )
+            self.rot_obj[ee].set_target_rotations(
+                wp.array(self.tgt_r[:, ee].astype(np.float32, copy=False), dtype=wp.vec4)
+            )
+
+        # Capture CUDA graph
+        if not (wp.get_device().is_cuda and wp.is_mempool_enabled(wp.get_device())):
+            raise SkipNotImplemented
```
[the remark this answers is not in the document]
Comment (Kenny-Vilella) [reply asv/benchmarks/simulation/bench_ik.py]: Done

File: asv/benchmarks/benchmark_ik.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+import newton.ik as ik
+import newton.utils
+
+
+def create_franka_model() -> newton.Model:
+    builder = newton.ModelBuilder()
+    builder.num_rigid_contacts_per_world = 0
+    builder.default_shape_cfg.density = 100.0
+    asset_path = newton.utils.download_asset("franka_emika_panda") / "urdf/fr3.urdf"
+    partial(newton.ModelBuilder.add_urdf, scale=1.0)(builder, asset_path, floating=False)
```
[the remark this answers is not in the document]
Comment (Kenny-Vilella) [reply asv/benchmarks/benchmark_ik.py]: Done

File: asv/benchmarks/simulation/bench_ik.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+            self.model,
+            q_best,
+            self.tgt_p,
+            self.tgt_r,
+            self.EE_LINKS,
+            self.POS_THRESH_M,
+            self.ORI_THRESH_RAD,
+        )
+        if not success.all():
+            n_failed = int((~success).sum())
```
Comment (adenzler-nvidia) [inline asv/benchmarks/simulation/bench_ik.py:98]: ASV runs `teardown()` *after* recording the timing result from `time_solve()`. If `teardown` raises, ASV logs the exception but still records the timing. This means a broken IK solver that produces garbage quickly would show "good" performance numbers with a teardown error buried in logs.

Consider moving the correctness check to a tracked metric instead:

```python
def track_success_rate(self, batch_size):
    q_best = self.winners_d.numpy()
    success = eval_success(
        self.solver, self.model, q_best,
        self.tgt_p, self.tgt_r, self.EE_LINKS,
        self.POS_THRESH_M, self.ORI_THRESH_RAD,
    )
    return float(success.mean())

track_success_rate.unit = "fraction"
```

This makes correctness a first-class ASV metric that gets tracked across commits and triggers regressions.

File: asv/benchmarks/benchmark_ik.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+###########################################################################
+
+from __future__ import annotations
+
+import numpy as np
+import warp as wp
+
+import newton
+import newton.ik as ik
+import newton.utils
```
Comment (adenzler-nvidia) [inline asv/benchmarks/benchmark_ik.py:30]: Every other benchmark utility module sets `wp.config.enable_backward = False` (e.g. `benchmark_mujoco.py:30`). Since the model is created with `requires_grad=False`, adding this here would be consistent and may improve kernel compilation time.

File: asv/benchmarks/benchmark_ik.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        lambda_factor=4.0,
+        jacobian_mode=ik.IKJacobianType.ANALYTIC,
+    )
+    return (
+        solver,
+        objectives[: len(ee_links)],
+        objectives[len(ee_links) : 2 * len(ee_links)],
+    )
+
+
```
Comment (adenzler-nvidia) [inline asv/benchmarks/benchmark_ik.py:91]: Both `fk_targets()` and `eval_success()` use `solver._fk_two_pass()`, which is a private method (accessed via `__getattr__` delegation). No other ASV benchmark relies on private APIs. If the IK internals are refactored, this silently breaks.

Worth adding a comment documenting the coupling, e.g.:
```python
# Uses internal _fk_two_pass for direct FK without full solve overhead.
# Update if IKSolver internals change.
```

File: asv/benchmarks/simulation/bench_ik.py [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+            self.solver.step(self.seeds_d, self.winners_d, iterations=self.ITERATIONS, step_size=self.STEP_SIZE)
+        self.solve_graph = cap.graph
+
+    NUM_SOLVES = 50
+
+    @skip_benchmark_if(wp.get_cuda_device_count() == 0)
+    def time_solve(self, batch_size):
+        for _ in range(self.NUM_SOLVES):
+            wp.capture_launch(self.solve_graph)
+        wp.synchronize_device()
```
Comment (adenzler-nvidia) [inline asv/benchmarks/simulation/bench_ik.py:83]: Nit: `NUM_SOLVES` is placed between `setup()` and `time_solve()`, but all other constants (`ITERATIONS`, `STEP_SIZE`, `SEED`, etc.) are grouped at the top of the class. The other benchmarks (e.g. `_FastBenchmark` in `bench_mujoco.py`) also keep all class-level constants together. Consider moving it up.

File: asv/benchmarks/simulation/bench_ik.py (hunk 6 above)
Comment (Kenny-Vilella) [reply asv/benchmarks/simulation/bench_ik.py:98]: This is incorrect, the error would not be buried into the log, it would fail the benchmark.

File: asv/benchmarks/benchmark_ik.py (hunk 7 above)
Comment (Kenny-Vilella) [reply asv/benchmarks/benchmark_ik.py:30]: Added.
But for the record it is not correct that "Every other benchmark" adds this, only a few.
It could make sense to add it everywhere, I need to check if it leads to any difference.

File: asv/benchmarks/benchmark_ik.py (hunk 8 above)
Comment (Kenny-Vilella) [reply asv/benchmarks/benchmark_ik.py:91]: I am not convinced this is really useful.
If the API change, then it will break and fail the benchmark, it's not like adding a comment would improve the situation.

File: asv/benchmarks/simulation/bench_ik.py (hunk 9 above)
Comment (Kenny-Vilella) [reply asv/benchmarks/simulation/bench_ik.py:83]: Done
